ingest/clusters.py

```diff
@@ -0,0 +1,80 @@
+import copy
+from typing import Dict, Generator, List, Tuple, Union
+
+from ingest_files import IngestFiles
+
+
+class Clusters(IngestFiles):
+
+    ALLOWED_FILE_TYPES = ['text/csv',
+                          'text/plain', 'text/tab-separated-values']
+    MAX_THRESHOLD = 100_000
+    SUBSAMPLE_THRESHOLDS = [MAX_THRESHOLD, 20000, 10000, 1000]
+    COLLECTION_NAME = 'clusters'
+    SUBCOLLECTION_NAME = 'data'
+
+    def __init__(self, file_path, *, name: str = None, study_accession: str = "", domain_ranges=None):
+
+        IngestFiles.__init__(self, file_path, self.ALLOWED_FILE_TYPES)
+        self.header = self.get_next_line(increase_line_count=False)
+        # Second line in cluster is metadata_type
+        self.metadata_types = self.get_next_line(increase_line_count=False)
+        self.unique_values = []
+        self.source_file_type = 'cluster'
+        # self.points = amount of rows
+        self.top_level_doc = {
+            'name': name,
+            'study_accession': study_accession,
+            'source_file_type': 'cluster',
+            'domain_ranges': domain_ranges,
+            'points': self.amount_of_lines,
+        }
+        self.annotation_subdocs = self.create_annotation_subdocs()
+        # sample size needs to be smaller than amount of points
+        # for each cell annotations if there are any
+
+    def update_points(self):
+        self.top_level_doc['points'] = self.amount_of_lines
+
+    def transform(self, rows):
+        """ Add data from cluster files into annotation subdocs in cluster data model"""
+        for idx, column in enumerate(rows):
```

Comment:
Could you add a brief comment in the code explaining what happens when `idx == 0`?  

I suspect your explanation from https://github.com/broadinstitute/scp-ingest-pipeline/pull/13#discussion_r309249856 applies here, but I won't remember that for long.

Reply:
Resolved in 60db95d